cg/meta/workflow/fluffy.py

```diff
@@ -0,0 +1,243 @@
+
+    def get_concentrations_from_lims(self, sample_id: str) -> str:
+        """Get sample concentration from LIMS"""
+        return self.lims_api.get_sample_attribute(lims_id=sample_id, key="concentration_sample")
+
+    def add_concentrations_to_samplesheet(
+        self, samplesheet_housekeeper_path: Path, samplesheet_workdir_path: Path
+    ) -> None:
+        csv_reader = csv.reader(open(samplesheet_housekeeper_path, "r"))
+        csv_writer = csv.writer(open(samplesheet_workdir_path, "w"))
+
+        sample_name_index = None
+        sample_id_index = None
+        for row in csv_reader:
+            if "SampleID" in row and not sample_id_index:
+                sample_name_index = row.index("SampleName")
+                sample_id_index = row.index("SampleID")
+                row.append("Library_nM")
+                csv_writer.writerow(row)
+                continue
+            if sample_id_index:
+                sample_id = row[sample_id_index]
+                row[sample_name_index] = self.get_sample_name_from_lims_id(lims_id=sample_id)
+                row.append(str(self.get_concentrations_from_lims(sample_id=sample_id)))
+                csv_writer.writerow(row)
+
+    def get_samplesheet_housekeeper_path(self, flowcell_name: str) -> Path:
+        return Path(
+            self.housekeeper_api.files(bundle=flowcell_name, tags=["samplesheet"])
+            .all()[0]
+            .full_path
+        )
+
+    def make_samplesheet(self, case_id: str, dry_run: bool) -> None:
+        """Create SampleSheet.csv file in working directory and add desired values to the file"""
+
+        case_obj = self.status_db.family(case_id)
+        flowcell_name = case_obj.links[0].sample.flowcells[0].name
+        samplesheet_housekeeper_path = self.get_samplesheet_housekeeper_path(
+            flowcell_name=flowcell_name
+        )
+        samplesheet_workdir_path = Path(self.get_samplesheet_path(case_id=case_id))
+        LOG.info(
+            "Writing modified csv from %s to %s",
+            samplesheet_housekeeper_path,
+            samplesheet_workdir_path,
+        )
+        if not dry_run:
+            self.add_concentrations_to_samplesheet(
+                samplesheet_housekeeper_path=samplesheet_housekeeper_path,
+                samplesheet_workdir_path=samplesheet_workdir_path,
+            )
+
+    def run_fluffy(self, case_id: str, dry_run: bool) -> None:
+        """Call fluffy with the configured command-line arguments"""
+
+        output_path: Path = self.get_output_path(case_id=case_id)
+        if output_path.exists():
+            LOG.info("Old working directory found, cleaning!")
+            shutil.rmtree(output_path, ignore_errors=True)
+        command_args = [
+            "--config",
+            self.fluffy_config.as_posix(),
+            "--sample",
+            self.get_samplesheet_path(case_id=case_id).as_posix(),
```

Comment:
As I understand it `.resolve()` is the recommended way to convert a `Path` object to a absolute path in `str` format. Or just `str(Path)` if not absolute. `as_posix` is just `str(self).replace(f.sep, '/')`